presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/AgentServerModule.java

```diff
@@ -40,7 +41,24 @@ protected void setup(Binder binder)
     {
         binder.disableCircularProxies();
 
+        // buildConfigObject also calls bindConfig
+        AgentConfig config = buildConfigObject(AgentConfig.class);
+        configBinder(binder).bindConfig(PrestoConfig.class, "presto");
+
+        switch (config.getPackageType()) {
```

Comment:
Look at `HiveSecurityModule`:
```
    @Override
    protected void setup(Binder binder)
    {
        bindSecurityModule("legacy", new LegacySecurityModule());
        bindSecurityModule("file", new FileBasedAccessControlModule());
        bindSecurityModule("read-only", new ReadOnlySecurityModule());
        bindSecurityModule("sql-standard", new SqlStandardSecurityModule());
    }

    private void bindSecurityModule(String name, Module module)
    {
        install(installModuleIf(
                SecurityConfig.class,
                security -> name.equalsIgnoreCase(security.getSecuritySystem()),
                module));
    }
```
You can do the same thing here if you put the RPM/tarball installation mechanisms in separate modules